packages/snaps-controllers/src/services/webview/WebviewExecutionService.ts

```diff
@@ -0,0 +1,25 @@
+import type { BasePostMessageStream } from '@metamask/post-message-stream';
+
+import type { ExecutionServiceArgs } from '../AbstractExecutionService';
+import { ProxyExecutionService } from '../proxy/ProxyExecutionService';
+
+type WebviewExecutionServiceArgs = {
+  stream: BasePostMessageStream;
+  frameUrl: URL;
+} & ExecutionServiceArgs;
+
+export class WebviewExecutionService extends ProxyExecutionService {
+  constructor({
+    messenger,
+    setupSnapProvider,
+    stream,
+    frameUrl,
+  }: WebviewExecutionServiceArgs) {
+    super({
+      messenger,
+      setupSnapProvider,
+      stream,
```

Comment:
We may need a stream for the execution service side that can consume the WebView and do both:
```
this.#targetWindow.injectJavaScript(`window.postMessage(${JSON.stringify(message)})`)
```

and handling of the messages received from the WebView. That piece seems to be missing right now